pkg/server/startup.go

```diff
@@ -54,18 +52,18 @@ func Startup(ctx context.Context, systemNamespace, kubeConfig string) error {
 	return nil
 }
 
-func bootstrapResources(rioContext *types.Context, systemNamespace string) (bool, error) {
+func bootstrapResources(rioContext *types.Context, systemNamespace string) error {
 	if _, err := rioContext.Apps.Apps().V1().Deployment().Get(systemNamespace, "rio-controller", metav1.GetOptions{}); errors.IsNotFound(err) {
+		constants.DevMode = true
```

Comment:
Hmm, not sure about this. 
If we can't find the controller we just enable dev mode? Seems concerning